File: Python/8kyu/8kyu-even-or-odd.py
# Even or Odd
# Difficulty: 8 kyu

# Solution
def even_or_odd(number):
    return "Even" if number % 2 == 0 else "Odd"

# Vice versa (shorter)
def even_or_odd(number):
    return "Odd" if number % 2 else "Even"
    
# Test

import codewars_test as test
from solution import even_or_odd

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(even_or_odd(2), "Even")
        test.assert_equals(even_or_odd(1), "Odd")
        test.assert_equals(even_or_odd(0), "Even")
        test.assert_equals(even_or_odd(1545452), "Even")
        test.assert_equals(even_or_odd(7), "Odd")
        test.assert_equals(even_or_odd(78), "Even")
        test.assert_equals(even_or_odd(17), "Odd")
        test.assert_equals(even_or_odd(74156741), "Odd")
        test.assert_equals(even_or_odd(100000), "Even")
        test.assert_equals(even_or_odd(-123), "Odd")
        test.assert_equals(even_or_odd(-456), "Even")

File: Julia/8kyu/8kyu-even-or-odd.jl
# Even or Odd
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

# Solution 1
module Solution
  export evenOrOdd
  function evenOrOdd(number)
    ifelse(number % 2 == 0, "Even", "Odd")
  end
end

# Solution 2 (can also use isodd() and reverse the outcomes)
module Solution
  export evenOrOdd
  function evenOrOdd(number)
    iseven(number) ? "Even" : "Odd"
  end
end


# Tests
using FactCheck
facts("Basic Tests") do
  @fact Solution.evenOrOdd(2) --> "Even"
  @fact Solution.evenOrOdd(1) --> "Odd"
  @fact Solution.evenOrOdd(0) --> "Even"
  @fact Solution.evenOrOdd(7) --> "Odd"
  @fact Solution.evenOrOdd(-2) --> "Even"
  @fact Solution.evenOrOdd(-1) --> "Odd"
end